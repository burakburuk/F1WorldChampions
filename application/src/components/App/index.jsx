import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import Bar from '../Bar';
import GlobalStyle from '../../assets/style/GlobalStyle';
import WorldChampionTableConnected from '../../containers/WorldChampionTableConnected';
import WinnersInYearDialogConnected from '../../containers/WinnersInYearDialogConnected';
import NotificationBoxConnected from '../../containers/NotificationBoxConnected';

const App = (props) => {
    const { store } = props;
    return (
        <Provider store={store}>
            <div>
                <GlobalStyle />
                <Bar/>
                <WorldChampionTableConnected />
                <WinnersInYearDialogConnected />
                <NotificationBoxConnected />
            </div>
        </Provider>
    );
};

App.propTypes = {
    store: PropTypes.object.isRequired,
};

export default App;
