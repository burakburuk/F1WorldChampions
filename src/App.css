.App {
    text-align: center;
}

.App-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 80px;
}

.App-header {
    background-color: #222;
    height: 150px;
    padding: 20px;
    color: white;
}

.App-title {
    font-size: 1.5em;
}

.App-intro {
    font-size: large;
}

.app-bar {
    width: 100%;
    height: 80px;
    background: #e00;
    color: white;
    vertical-align: middle;
}

.app-bar img {
    width: 120px;
    margin-left: 18px;
    margin-top: 25px;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}

.list-header {
    width: 800px;
    margin-left: auto;
    margin-right: auto;
    color: #4c4c4c;
}

.list-header-driver {
    width: 33%;
    padding-left: 80px;
}

.list-header-nationality {
    width: 18%;
}

.list-header-team {
    width: 33%;
}

.list-header-points {

}

.ch-item {
    color: gray;
    text-align: center;
    padding-top: 2px;
}

.by-item {
    color: gray;
}

.by-driver {
    color: black;
}

.by-name-item-highlight {
    color: #ec7373;
}

.item-year {
    transform: rotate(270deg);
    width: 54px;
    height: 27px;
    text-align: center;
    margin-left: -25px;
    margin-top: -14px;
    background: #bfbfbf;
    color: white;
    padding-top: 11px;
    border-right: 10px solid red;
}

.item-driver {
    color: #000000;
}

.item-nationality {

}

.item-company {
    text-align: right;
}

.item-points {
    text-align: right;
}

.item-image {
    text-align: center;
    padding-top: 0;
}

.item-race {
    font-size: 18px;
}

.item-image img {
    width: 60%;
}
