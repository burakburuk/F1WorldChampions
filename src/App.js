import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux'
import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
      const {store} = this.props;
    return (
        <Provider store={store}>
            <div className="App">
                <header className="App-header">
                    <img src={logo} className="App-logo" alt="logo" />
                    <h1 className="App-title">Welcome to React</h1>
                </header>
                <p className="App-intro">
                    To get started, edit <code>src/App.js</code> and save to reload.
                </p>
            </div>
        </Provider>
    );
  }
}

App.propTypes = {
    store: PropTypes.object.isRequired
};

export default App;
