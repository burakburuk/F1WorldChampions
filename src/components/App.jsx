import '../index.scss';

import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import Bar from './Bar';
import WorldChampionTableContainer from '../container/WorldChampionTableContainer';
import WinnersInYearDialogContainer from '../container/WinnersInYearDialogContainer';
import NotificationBoxContainer from '../container/NotificationBoxContainer';

const App = (props) => {
    const { store } = props;
    return (
        <Provider store={store}>
            <div>
                <Bar/>
                <WorldChampionTableContainer />
                <WinnersInYearDialogContainer />
                <NotificationBoxContainer />
            </div>
        </Provider>
    );
};

App.propTypes = {
    store: PropTypes.object.isRequired,
};

export default App;
