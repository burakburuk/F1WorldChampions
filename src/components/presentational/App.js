import '../../index.scss';

import React from 'react';
import PropTypes from 'prop-types';
import {Provider} from 'react-redux';
import WorldChampions from './WorldChampions';
import Bar from './Bar';

const App = (props) => {
    const {store} = props;
    return (
        <Provider store={store}>
            <div>
                <Bar/>
                <WorldChampions></WorldChampions>
            </div>
        </Provider>
    );
};

App.propTypes = {
    store: PropTypes.object.isRequired
};

export default App;
