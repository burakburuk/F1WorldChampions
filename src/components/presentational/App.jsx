import '../../index.scss';

import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import WorldChampions from './WorldChampions';
import ChampionsPopup from '../container/ChampionsPopup';
import Bar from './Bar';

const App = (props) => {
    const { store } = props;
    return (
        <Provider store={store}>
            <div className={'App'}>
                <Bar/>
                <WorldChampions></WorldChampions>
                <ChampionsPopup />
            </div>
        </Provider>
    );
};

App.propTypes = {
    store: PropTypes.object.isRequired,
};

export default App;
