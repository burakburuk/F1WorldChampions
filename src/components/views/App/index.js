import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {Provider} from 'react-redux'
import WorldChampions from '../WorldChampions';
import Bar from "../Bar";
import '../../../index.css';

class App extends Component {
    render() {
        const {store} = this.props;
        return (
            <Provider store={store}>
                <div>
                    <Bar/>
                    <WorldChampions></WorldChampions>
                </div>
            </Provider>
        );
    }
}

App.propTypes = {
    store: PropTypes.object.isRequired
};

export default App;
