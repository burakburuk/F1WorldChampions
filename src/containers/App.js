import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux'
import ChampionCard from '../components/Champions/ChampionList';
import '../App.css';

class App extends Component {
  render() {
      const {store} = this.props;
    return (
        <Provider store={store}>
            <ChampionCard></ChampionCard>
        </Provider>
    );
  }
}

App.propTypes = {
    store: PropTypes.object.isRequired
};

export default App;
