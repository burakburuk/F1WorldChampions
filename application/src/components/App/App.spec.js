import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store'
import App from './';

const mockStore = configureStore();
const store = mockStore();

it('should match with snapshot', () => {
    const wrapper = shallow(<App store={store} />);
    expect(wrapper).toMatchSnapshot();
});
