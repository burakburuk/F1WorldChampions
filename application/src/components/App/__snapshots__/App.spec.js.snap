// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div>
    <GlobalStyleComponent />
    <Bar />
    <Connect(WorldChampionTableContainer) />
    <Connect(WinnersInYearDialogContainer) />
    <Connect(NotificationBoxContainer) />
  </div>
</Provider>
`;
